### ISSUE TITLE
convention when input can be an instance of a class

### ISSUE THREAD

**Issue by cmeeren** [numpy/numpydoc#43]
What is the convention for parameter type if a parameter is an instance of specific class, e.g. datetime.datetime?

Specifically, I have a parameter which can be either a float, datetime.date, or datetime.datetime. How should I write this in the parameter section?

* my_param : float or date or datetime
* my_param : float or datetime.date or datetime.datetime
* my_param : float or :class:\`datetime.date\` or :class:\`datetime.datetime\`
* my_param : float or :class:\`~datetime.date\` or :class:\`~datetime.datetime\`
* something else?

And of course, maybe a comma instead of the first "or", but that's nitpicking (which I like by the way) and not too important in this context (unless I'm mistaken).

**Comment by Eric89GXL** [comment]
I don't think that they currently get linked automatically. I have done a little bit of testing, and this sort of thing seems to work to generate links to `MyClass` and `MyOtherClass` docs:
```
my_param : float or :class:`MyClass` or :class:`MyOtherClass`
```
It would be nice if they could be auto-detected, though. I wonder if actually we could just try to auto-link them all with `:class:` and see what happens.

**Comment by Eric89GXL** [comment]
For a bad first pass at how this could be automated see #57
